### PR TITLE
Point cloud parameter for Processing

### PR DESCRIPTION
**Pull request by alexbruy**
## Description
Add a new point cloud parameter to Processing. Also allow to filter point cloud layers in map layer and multiple layers parameters. This opens the door for native point cloud algorithms and improves point cloud support in 3rd party providers.

### CODE REVIEW

File: tests/src/analysis/testqgsprocessing.cpp [hunk 1]
```diff
@@ -8998,6 +9000,135 @@ void TestQgsProcessing::parameterDxfLayers()
...
 
+void TestQgsProcessing::parameterPointCloudLayer()
+{
+  // setup a context
+  QgsProject p;
+  p.setCrs( QgsCoordinateReferenceSystem::fromEpsgId( 28353 ) );
+  QString testDataDir = QStringLiteral( TEST_DATA_DIR ) + '/'; //defined in CmakeLists.txt
+  QString vector1 = testDataDir + "multipoint.shp";
+  QString raster = testDataDir + "landsat.tif";
+  QString pointCloud = testDataDir + "point_clouds/las/cloud.las";
```
Comment (nyalldawson) [inline tests/src/analysis/testqgsprocessing.cpp:9037]: Better to use the ept layers here -- they don't require the pdal dependency

File: tests/src/analysis/testqgsprocessing.cpp [hunk 2]
```diff
@@ -8998,6 +9000,135 @@ void TestQgsProcessing::parameterDxfLayers()
   QCOMPARE( dxfList.at( 0 ).layerOutputAttributeIndex(), dxfLayer.layerOutputAttributeIndex() );
 }
 
+void TestQgsProcessing::parameterPointCloudLayer()
```
Comment (nyalldawson) [inline tests/src/analysis/testqgsprocessing.cpp:9029]: See below, but this should be ifdefed only if WITH_EPT

File: tests/src/analysis/testqgsprocessing.cpp [hunk 3]
```diff
@@ -860,10 +860,16 @@ void TestQgsProcessing::compatibleLayers()
   QgsMeshLayer *m2 = new QgsMeshLayer( fm.filePath(), "mA", "mdal" );
   QVERIFY( m2->isValid() );
```
Comment (nyalldawson) [inline tests/src/analysis/testqgsprocessing.cpp:862]: Again, this should use ept files and be conditional on WITH_EPT

File: tests/src/gui/testprocessinggui.cpp [hunk 4]
```diff
@@ -9899,6 +9900,149 @@ void TestProcessingGui::testMeshDatasetWrapperLayerOutsideProject()
 
 }
 
+void TestProcessingGui::testPointCloudLayerWrapper()
+{
+  // setup a project with a range of layer types
+  QgsProject::instance()->removeAllMapLayers();
+  QgsPointCloudLayer *cloud1 = new QgsPointCloudLayer( QStringLiteral( TEST_DATA_DIR ) + "/point_clouds/las/cloud.las", QStringLiteral( "cloud1" ), QStringLiteral( "pdal" ) );
```
Comment (nyalldawson) [inline tests/src/gui/testprocessinggui.cpp:9907]: Also ept files and conditional

**Comment by nyalldawson** [comment]
Nice work!

File: tests/src/analysis/testqgsprocessing.cpp (hunk 1 above)
Comment (alexbruy) [reply tests/src/analysis/testqgsprocessing.cpp]: Thanks, fixed.

File: tests/src/analysis/testqgsprocessing.cpp (hunk 2 above)
Comment (alexbruy) [reply tests/src/analysis/testqgsprocessing.cpp:9046]: Done.

File: tests/src/analysis/testqgsprocessing.cpp (hunk 3 above)
Comment (alexbruy) [reply tests/src/analysis/testqgsprocessing.cpp:864]: Done

File: tests/src/gui/testprocessinggui.cpp (hunk 4 above)
Comment (alexbruy) [reply tests/src/gui/testprocessinggui.cpp]: Also fixed.

**Review by nyalldawson — APPROVED** [review]

**Pull request merged**